node/resolvers/Queries/CostCenters.ts

```diff
@@ -287,28 +287,49 @@ const costCenters = {
     ctx: Context
   ) => {
     const {
-      clients: { masterdata },
+      clients: { masterdata, storefrontPermissions },
       vtex: { logger, sessionData },
     } = ctx as any
 
     // create schema if it doesn't exist
     await checkConfig(ctx)
 
-    if (!sessionData?.namespaces['storefront-permissions']) {
-      throw new GraphQLError('organization-data-not-found')
-    }
-
     const {
-      organization: { value: userOrganizationId },
-    } = sessionData.namespaces['storefront-permissions']
+      data: { checkUserPermission },
+    }: any = await storefrontPermissions
+      .checkUserPermission('vtex.b2b-orders-history@0.x')
+      .catch((error: any) => {
+        logger.error({
+          message: 'checkUserPermission-error',
+          error,
+        })
+
+        return {
+          data: {
+            checkUserPermission: null,
+          },
+        }
+      })
 
-    if (!id) {
-      // get user's organization from session
-      id = userOrganizationId
-    }
+    const isSalesAdmin = checkUserPermission?.role.slug.match(/sales-admin/)
```

Comment:
Ideally, rather than checking their role, we should check if they have a specific permission. This would require adding a new permission to b2b-organizations like "View Cost Centers In Any Organization", so perhaps we can put this on the backlog.